File: frontend/src/store/modules/migration.ts
import axios from "axios";
import { ConnectionInfo, InstanceMigration, SqlResultSet } from "../../types";

const getters = {};

const actions = {
  async checkMigrationSetup(
    {}: any,
    connectionInfo: ConnectionInfo
  ): Promise<InstanceMigration> {
    const data = (
      await axios.post(`/api/migration/instance/status`, {
        data: {
          type: "connectionInfo",
          attributes: connectionInfo,
        },
      })
    ).data.data;

    return {
      status: data.attributes.status,
      error: data.attributes.error,
    };
  },

  async createkMigrationSetup(
    { rootGetters }: any,
    connectionInfo: ConnectionInfo
  ): Promise<SqlResultSet> {
    const data = (
      await axios.post(`/api/migration/instance`, {
        data: {
          type: "connectionInfo",
          attributes: connectionInfo,
        },
      })
    ).data.data;

    return rootGetters["sql/convert"](data);
  },
};

const mutations = {};

export default {
  namespaced: true,
  getters,
  actions,
  mutations,
};

File: api/migration.go
package api

// Instance migration status
type InstanceMigrationStatus string

const (
	InstanceMigrationUnknown  InstanceMigrationStatus = "UNKNOWN"
	InstanceMigrationOK       InstanceMigrationStatus = "OK"
	InstanceMigrationNotExist InstanceMigrationStatus = "NOT_EXIST"
)

func (e InstanceMigrationStatus) String() string {
	switch e {
	case InstanceMigrationUnknown:
		return "UNKNOWN"
	case InstanceMigrationOK:
		return "OK"
	case InstanceMigrationNotExist:
		return "NOT_EXIST"
	}
	return "UNKNOWN"
}

type InstanceMigration struct {
	Status InstanceMigrationStatus `jsonapi:"attr,status"`
	Error  string                  `jsonapi:"attr,error"`
}
